content/docs/practices/remote_write.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Remote write tuning
+sort_rank: 8
+---
+
+# Remote write tuning
+
+Prometheus implements sane defaults for remote write, but many users have
+different requirements and would like to optimize their remote settings.
+
+This page describes the tuning parameters available via the [remote write
+configuration.](/docs/prometheus/latest/configuration/configuration/#remote_write)
+
+## Remote write characteristics
+
+Each remote write destination starts a queue which reads from the write-ahead
+log (WAL), writes the samples into an in memory queue owned by a shard, which
+then sends a request to the configured endpoint. The flow of data looks like:
+
+```
+      |-->  queue (shard_1)   --> remote endpoint
+WAL --|-->  queue (shard_...) --> remote endpoint
+      |-->  queue (shard_n)   --> remote endpoint
+```
+
+When one shard backs up and fills its queue, Prometheus will block reading from
+the WAL into any shards. Failures will be retried without loss of data unless
+the remote endpoint remains down for more than 2 hours. After 2 hours, the WAL
+will be compacted and data that has not been sent will be lost.
+
+During operation, Prometheus will continuously calculate the optimal number of
+shards to use based on the incoming sample rate, number of outstanding samples
+not sent, and time taken to send each sample.
+
+### Memory usage
+
+Using remote write increases the memory footprint of Prometheus. Most users
+report ~25% increased memory usage, but that number is dependent on the shape
+of the data. For each series in the WAL, the remote write code caches a mapping
+of series ID to label values, causing large amounts of series churn to
+significantly increase memory usage.
+
+Memory usage will also increase based on the number of shards used, which can be
+capped by setting `max_shards`.
+
+## Parameters
+
+All the relevant parameters are found under the `queue_config` section of the
+remote write configuration.
+
+### `capacity`
+
+Capacity controls how many samples are queued in memory per shard before
+blocking reading from the WAL. Once the WAL is blocked, samples cannot be
+appended to any shards and all throughput will cease.
+
+Capacity should be high enough to avoid blocking other shards in most
+cases, but too much capacity can cause excess memory consumption and longer
+times to clear queues during resharding. It is recommended to set capacity 
+to 3-10 times `max_samples_per_send`.
+
+### `max_shards`
+
+Max shards configures the maximum number of shards, or parallelism, Prometheus
+will use for each remote write queue. Prometheus will try not to use too many
+shards, but if the queue falls behind the remote write component will increase
+the number of shards up to max shards to increase thoughput.
```

Comment:
Perhaps add something about the memory impact (as the number of shardes multiplies with the capacity to determine how many samples are queued at max). Also, [there were reports of lock contention with too many shards](https://github.com/prometheus/prometheus/issues/5803#issuecomment-520973662). Perhaps give some sane ballpark numbers here so that people don't set this to thousands of shards?

Reply:
I added more information about the memory impact of shards. I did not add anything specific about lock contention, but did say that it is unlikely to need to increase the default value for `max_shards`.